### DISCUSSION TITLE
Field resolver not working properly

### DISCUSSION THREAD

**Discussion by lemiesz** [MichalLytek/typegraphql-prisma#335] [Q&A]
**Describe the Bug**
For some reason when adding @Root to a field resolver it ceases to work properly

**To Reproduce**

I have the following model 
 
 _Warning some elements are simplified/redacted both for privacy reasons and to make example easier to illustrate _ 
 
```
@ObjectType()
export class Rule {
    constructor({ , ruleId }: Rule) {
        return {

            ruleId,
        };
    }

    @Field(() => ID)
    ruleId: string;

    ruleTree?: string;
}
```
and the associated resolver 

```
@Resolver((of) => Rule)
export default class RuleResolver implements ResolverInterface<Rule> {


    @FieldResolver((of) => String)
    public async ruleTree(@Root() root: Rule) {
        return 'Some nonsense';
    }
}

``` 

I think attempt to run the following query 
 
 
 ```
 {
  parentObject(input:{
    id: "id",
  }) {
    id,
    rootRule {
      ruleId,
      parentLearningPathId,
       ruleTree
    }
}
}
 ``` 
 

 Rule is a model that exists on parentObject. 
 
 When I write my field resolver as listed above it just never runs and I get a `non-nullable field` error. 
 
 When I write my field resolver without adding in the @Root decorator. Then it properly returns `some nonsense` as the ruleTrue field.

**Comment by lemiesz** [comment]
To reiterate this odes not work

```
@Resolver((of) => Rule)
export default class RuleResolver implements ResolverInterface<Rule> {


    @FieldResolver((of) => String)
    public async ruleTree(@Root() root: Rule) {
        return 'Some nonsense';
    }
}
``` 

While this does 

```
@Resolver((of) => Rule)
export default class RuleResolver implements ResolverInterface<Rule> {


    @FieldResolver((of) => String)
    public async ruleTree() {
        return 'Some nonsense';
    }
}
```

**Comment by MichalLytek** [comment]
Object type classes can't have parameters in constructor.

**Comment by lemiesz** [comment]
Hmmm its a common pattern I have been following that has worked fine. However let me try that and see if it fixes this issue

**Comment by MichalLytek** [comment]
You're also returning an object from the constructor, which breaks `this` and inheritance chain or methods access.
